### PR TITLE
Add basic support for custom random distributions

### PR DESCRIPTION
**Pull request by marcin-golebiowski**

### CODE REVIEW

**Comment by marcin-golebiowski** [comment]
It's not that simple to generate random number with custom distribution. I will fight it today again :)

File: src/SpiceSharpParser/ModelReaders/Netlist/Spice/Context/Models/PercentSpecification.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using SpiceSharpParser.Models.Netlist.Spice.Objects;
+
+namespace SpiceSharpParser.ModelReaders.Netlist.Spice.Context.Models
+{
+    public class PercentSpecification
```
Comment (svenboulanger) [inline src/SpiceSharpParser/ModelReaders/Netlist/Spice/Context/Models/PercentSpecification.cs]: What does this class do? Is "Percent" relevant to its function, or is it used as a general description of a distribution specification (in which case the name is too narrow)?

File: src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public CustomRandomNumberProvider(Pdf pdf, IRandomDoubleProvider baseRandom)
             _baseRandom = baseRandom;
         }
 
-        public double NextDouble()
+        public double NextSignedDouble()
```
Comment (svenboulanger) [inline src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs]: What is this function _supposed_ to return? I can see that it gives a value bound by `int.MaxValue` but what's the ideal behavior?

File: src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs [hunk 3]
```diff
@@ -6,6 +6,8 @@ public interface IRandomNumberProvider : IRandomIntegerProvider, IRandomDoublePr
 
     public interface IRandomDoubleProvider
     {
+        double NextSignedDouble();
```
Comment (svenboulanger) [inline src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs]: How does this method differ from `NextDouble` for a _general_ random number provider?

File: src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs (hunk 2 above)
Comment (marcin-golebiowski) [reply src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs]: @svenboulanger `Next` should return random integer from 0 to int.max. 
I'm aware that current implementation `int.MaxValue * NextDouble()` is very far away from perfect.

File: src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs (hunk 3 above)
Comment (marcin-golebiowski) [reply src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs]: @svenboulanger  I've found in PSpice and other documation that  .DISTRIBUTION dot statement specifies random number that in in range (-1,1). I would like to have method to easy generate number in that range.

File: src/SpiceSharpParser/ModelReaders/Netlist/Spice/Context/Models/PercentSpecification.cs (hunk 1 above)
Comment (marcin-golebiowski) [reply src/SpiceSharpParser/ModelReaders/Netlist/Spice/Context/Models/PercentSpecification.cs]: @marcin-golebiowski I've changed the name of the class to `ParameterRandomness`. It specifies what randomness to use for that parameter.

File: src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs (hunk 2 above)
Comment (svenboulanger) [reply src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs]: I'm not saying it's far from perfect. It just seems pretty arbitrary to use `int.MaxValue`. Why would this need to be `int.MaxValue` and not a double between 0 and 1? Why not `double.MaxValue`? What is it going to be used for?

File: src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs (hunk 3 above)
Comment (svenboulanger) [reply src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs]: Then is a number generator between (0,1) still necessary? And does it have to be produced in the same `IRandomDoubleProvider` or is it more logical to have *two* random double providers: one giving you (-1, 1) and one producing (0, 1)?

File: src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs (hunk 2 above)
Comment (marcin-golebiowski) [reply src/SpiceSharpParser/Common/Mathematics/Probability/CustomRandomNumberProvider.cs]: @svenboulanger We are talking about `Next` method that is also in standard `Random` class. 
I wanted to have something similar in custom random provider. If it doesn't make sense, I will remove it :)

File: src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs (hunk 3 above)
Comment (marcin-golebiowski) [reply src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs]: @svenboulanger  I think it's not necessary at this moment to split it into two providers. 
The changes that I want to commit it's the first step in providing good randomness support in SpiceSharpParser.

Comment (marcin-golebiowski) [reply src/SpiceSharpParser/Common/Mathematics/Probability/IRandomNumberProvider.cs]: @svenboulanger  There will be refactor for sure in the near future.

**Comment by marcin-golebiowski** [comment]
@svenboulanger  Can I merge it?

**Comment by marcin-golebiowski** [comment]
@svenboulanger I will merge it, but I will refactor code in the near future.

**Pull request merged**